lib/puppetserver/ca/actions/clean.rb

```diff
@@ -0,0 +1,159 @@
+require 'puppetserver/ca/utils/parse'
+require 'puppetserver/ca/utils/http_client'
+require 'puppetserver/ca/utils/file_system'
+require 'puppetserver/ca/config/puppet'
+require 'puppetserver/ca/actions/revoke'
+
+require 'optparse'
+require 'json'
+
+module Puppetserver
+  module Ca
+    module Actions
```

Comment:
For namespacing like this, should it be `Action` or `Actions`?

Reply:
I think the coherence of the object within the namespace helps. Like I think of `Utils::FileSystem` as a collection of utility methods for interacting with the filesystem. That kinda breaks down with `HttpClient` as a singular entity vs a collection of methods. With `Actions::Clean` I would expect to be a collection of methods for cleaning vs `Action::Clean` as a singular clean action. I guess its really that I think of the parent module as providing some description of the child object vs describing itself...

tl;dr: I'd go with `Action`, and I could go either way on `Utils`.